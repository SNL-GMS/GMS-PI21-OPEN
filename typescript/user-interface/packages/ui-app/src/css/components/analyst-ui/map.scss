@import '../../gms-fonts.scss';

.map {
  width: 100%;
  height: 100%;
  padding: 0.5rem;
}
.map__status-online-indicator {
  position: absolute;
  bottom: 0.8rem;
  right: 1rem;
  color: var(--gms-main, $gms-main);
  .map__status-online-indicator__online-mode {
    color: var(--gms-main, $gms-main);
    text-shadow: -1px 0 var(--gms-recessed, $gms-recessed), 0 1px var(--gms-recessed, $gms-recessed),
      1px 0 var(--gms-recessed, $gms-recessed), 0 -1px var(--gms-recessed, $gms-recessed);
  }
  .map__status-online-indicator__offline-mode {
    color: $gms-strong-warning;
    text-shadow: -1px 0 var(--gms-recessed, $gms-recessed), 0 1px var(--gms-recessed, $gms-recessed),
      1px 0 var(--gms-recessed, $gms-recessed), 0 -1px var(--gms-recessed, $gms-recessed);
  }
}
.map__column {
  display: flex;
  height: 100%;
  flex-direction: column;
}
.map__row {
  flex: 1 1 auto;
  display: flex;
}
.map-controls {
  flex: 0 0 auto;
  display: flex;
  align-items: center;
  padding-left: 0.5rem;
}
.map-controls__label {
  margin-bottom: 0rem !important;
  margin-left: 1rem !important;
}
.map-layer-toggles {
  flex: 0 0 auto;
  width: 130px;
  display: flex;
  flex-direction: column;
  background-color: var(--gms-prominent-background, $gms-prominent-background);
}
.map__inner-column {
  flex: 1 1 auto;
  border-left: 1px solid var(--gms-background, $gms-background);
  display: flex;
  flex-direction: column;
}
.map__rendering-wrapper-1 {
  flex: 1 1 auto;
  position: relative;
}

.map__layer-button {
  position: absolute;
  z-index: 10;
  top: 5px; //to align with cesium toolbar buttons
  left: 5px;
}

.map-time {
  font-family: $gms-mono;
}

.ian-map-gl-container {
  height: 100%;
  padding: 0;
}

.layer-panel-content {
  &__container {
    margin-top: -28px;
    margin-left: 10px;
    position: absolute;
    background-color: var(--gms-popover-background, $gms-popover-background);
  }
  &--header {
    padding: 3px 8px 8px 12px;
    border-bottom: solid 1px;
  }
  &--checkbox-list {
    margin-left: 8px;
  }
}

// Disabling Columbus view
.cesium-sceneModePicker-dropDown-icon[data-bind*='COLUMBUS_VIEW'] {
  display: none;
}

.sd-swatch-tooltip {
  padding: 5px 6px;
  background-color: var(--gms-table-header-background, $gms-table-header-background);
  color: #bfccd6;
  border-radius: 3px;
  box-shadow: 0 0 0 1px rgba(16, 22, 26, 10%), 0 2px 4px rgba(16, 22, 26, 20%),
    0 8px 24px rgba(16, 22, 26, 20%);
}
