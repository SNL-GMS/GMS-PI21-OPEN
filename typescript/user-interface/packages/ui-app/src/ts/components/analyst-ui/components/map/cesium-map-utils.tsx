import type { ChannelTypes, StationTypes } from '@gms/common-model';
import { secondsToString, TIME_FORMAT_WITH_FRACTIONAL_SECOND_PRECISION } from '@gms/common-util';
import type {
  ColorMaterialProperty,
  ConstantPositionProperty,
  DistanceDisplayCondition
} from 'cesium';
import {
  BillboardGraphics,
  Cartesian3,
  Color,
  ConstantProperty,
  HorizontalOrigin,
  JulianDate,
  LabelGraphics,
  LabelStyle,
  PolylineGraphics,
  PolylineOutlineMaterialProperty,
  VerticalOrigin
} from 'cesium';

import {
  alwaysDisplayDistanceDisplayCondition,
  LABEL_HEIGHT_SELECTED,
  LABEL_HEIGHT_UNSELECTED,
  SELECT_BLUE as SELECTED_BLUE,
  SELECT_TRANSPARENT as SELECTED_TRANSPARENT,
  SELECTED_SIGNAL_DETECTION_GLOW_ALPHA,
  SELECTED_SIGNAL_DETECTION_GLOW_OUTLINE_WIDTH,
  SELECTED_SIGNAL_DETECTION_WIDTH_OFFSET
} from '~common-ui/components/map/constants';
// TODO move constants to common
import {
  fontStyle,
  imageScale,
  imageScaleSelected,
  monoFontStyle,
  selectedPixelOffset,
  unselectedPixelOffset
} from '~data-acquisition-ui/components/soh-map/constants';

import type { MapEventSource } from './types';

/**
 * Returns polyline material for Signal Detections based on whether or not the signal detection is selected or not.
 * If the Polyline is not selected, returns just the color
 * If the SD is selected, returns a new PolylineMaterialProperty that includes a semi-transparent outline to give the line a glow effect.
 *
 * @param isSelected
 * @param color
 */
export function getSDPolylineMaterial(
  isSelected: boolean,
  color: ColorMaterialProperty
): ColorMaterialProperty | PolylineOutlineMaterialProperty {
  if (!isSelected) return color;

  const outlineColor = color.color.getValue(JulianDate.now());
  outlineColor.alpha = SELECTED_SIGNAL_DETECTION_GLOW_ALPHA;
  return new PolylineOutlineMaterialProperty({
    color: color.color,
    outlineWidth: SELECTED_SIGNAL_DETECTION_GLOW_OUTLINE_WIDTH,
    outlineColor
  });
}

/**
 * Given a Cesium entity and a display condition, configure a label for that entity
 *
 * @param item - Station/Channel Group/Event location we are creating a label for
 * @param distanceDisplayCondition - DistanceDisplayCondition
 * @param isSelected - is the entity we are creating a label for selected (affects size and color)
 */
export function createLabel(
  item: ChannelTypes.ChannelGroup | StationTypes.Station | MapEventSource,
  distanceDisplayCondition: DistanceDisplayCondition,
  isSelected: boolean
): LabelGraphics {
  let text;
  let font;
  let backgroundColor;
  if ('name' in item) {
    text = item.name;
    font = fontStyle;
    backgroundColor = SELECTED_BLUE;
  } else {
    text = secondsToString(item.time, TIME_FORMAT_WITH_FRACTIONAL_SECOND_PRECISION);
    font = monoFontStyle;
    backgroundColor = SELECTED_TRANSPARENT;
  }
  const options: LabelGraphics.ConstructorOptions = {
    backgroundColor,
    text,
    font,
    fillColor: Color.WHITE,
    outlineColor: Color.BLACK,
    outlineWidth: 2,
    showBackground: isSelected,
    style: new ConstantProperty(LabelStyle.FILL_AND_OUTLINE),
    distanceDisplayCondition: new ConstantProperty(
      isSelected ? alwaysDisplayDistanceDisplayCondition : distanceDisplayCondition
    ),
    verticalOrigin: new ConstantProperty(VerticalOrigin.TOP),
    pixelOffset: new ConstantProperty(isSelected ? selectedPixelOffset : unselectedPixelOffset),
    eyeOffset: new Cartesian3(
      0.0,
      0.0,
      isSelected ? LABEL_HEIGHT_SELECTED : LABEL_HEIGHT_UNSELECTED
    )
  };

  return new LabelGraphics(options);
}

/**
 * Returns a new PolyLineGraphics object from an array of positions with given style
 *
 * @param positions Start position and end position as a Cesium.Cartiesian3[]
 * @param distanceDisplayCondition defines what distances you can see the line from
 * @param material If just specifying a color, use ColorMaterialProperty. If you want a line with outline provide a PolylineMaterialProperty
 * @param width of the polyline in pixels
 * @param isSelected
 */
export function createPolyline(
  positions: Cartesian3[],
  distanceDisplayCondition: DistanceDisplayCondition,
  material: ColorMaterialProperty | PolylineOutlineMaterialProperty,
  width: number,
  isSelected = false
): PolylineGraphics {
  return new PolylineGraphics({
    distanceDisplayCondition: new ConstantProperty(distanceDisplayCondition),
    positions,
    show: true,
    width: isSelected
      ? new ConstantProperty(width + (SELECTED_SIGNAL_DETECTION_WIDTH_OFFSET as number))
      : new ConstantProperty(width),
    material
  });
}

/**
 * Create a Billboard for a station, ChannelGroup, or event (i.e. map icons)
 *
 * @param selected : is the entity we are creating a billboard for selected
 * @param eyeOffset: z-index of the billboard
 * @param color
 * @param image
 */
export function createBillboard(
  selected: boolean,
  eyeOffset: ConstantPositionProperty,
  color: Color,
  image: Document | string
): BillboardGraphics {
  const billboard = new BillboardGraphics();
  billboard.image = new ConstantProperty(image);
  billboard.color = new ConstantProperty(color);
  billboard.scale = selected
    ? new ConstantProperty(imageScaleSelected)
    : new ConstantProperty(imageScale);
  // billboard should have the center of lat/long in middle of shape;
  billboard.horizontalOrigin = new ConstantProperty(HorizontalOrigin.CENTER);
  billboard.verticalOrigin = new ConstantProperty(VerticalOrigin.CENTER);
  // set eye offset (similar to z-index, which is not yet supported)
  billboard.eyeOffset = eyeOffset;
  return billboard;
}
