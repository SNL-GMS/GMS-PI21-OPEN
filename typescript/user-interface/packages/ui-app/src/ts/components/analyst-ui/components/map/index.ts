export { IANMap } from './ian-map-component';
