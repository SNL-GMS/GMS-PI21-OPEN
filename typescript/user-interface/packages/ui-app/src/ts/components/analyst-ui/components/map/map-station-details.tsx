import { ContextMenu } from '@blueprintjs/core';
import { Form, FormTypes } from '@gms/ui-core-components';
import React from 'react';

import { getTableCellStringValue } from '~common-ui/common/table-utils';

export interface IANStationDetailsProps {
  stationName: string;
  latitude: string;
  longitude: string;
  elevation: string;
  detailedType: string;
  entityType: string;
}

/**
 * MapStationDetails Component
 */
// eslint-disable-next-line react/function-component-definition
export const MapStationDetails: React.FunctionComponent<IANStationDetailsProps> = (
  props: IANStationDetailsProps
) => {
  const { stationName, latitude, longitude, elevation, detailedType, entityType } = props;

  const formItems: FormTypes.FormItem[] = [];
  formItems.push({
    itemKey: 'Name',
    labelText: 'Name',
    itemType: FormTypes.ItemType.Display,
    displayText: `${getTableCellStringValue(stationName)}`,
    displayTextFormat: FormTypes.TextFormats.Time
  });
  formItems.push({
    itemKey: 'Lat (째)',
    labelText: 'Lat (째)',
    itemType: FormTypes.ItemType.Display,
    displayText: `${getTableCellStringValue(latitude)}`,
    // Apply text format below to get monospace typeface per UX
    displayTextFormat: FormTypes.TextFormats.Time
  });
  formItems.push({
    itemKey: 'Lon (째)',
    labelText: 'Lon (째)',
    itemType: FormTypes.ItemType.Display,
    displayText: `${getTableCellStringValue(longitude)}`,
    // Apply text format below to get monospace typeface per UX
    displayTextFormat: FormTypes.TextFormats.Time
  });
  formItems.push({
    itemKey: 'Elevation (km)',
    labelText: 'Elevation (km)',
    itemType: FormTypes.ItemType.Display,
    displayText: `${getTableCellStringValue(elevation)}`,
    // Apply text format below to get monospace typeface per UX
    displayTextFormat: FormTypes.TextFormats.Time
  });
  if (entityType === 'Station') {
    formItems.push({
      itemKey: 'Type',
      labelText: 'Type',
      itemType: FormTypes.ItemType.Display,
      displayText: `${getTableCellStringValue(detailedType)}`
    });
  }

  const defaultPanel: FormTypes.FormPanel = {
    formItems,
    name: 'Additional Details'
  };

  const headerText = entityType === 'Station' ? 'Station Details' : 'Site Details';

  return (
    <div className="map-station-details__container">
      <Form
        header={headerText}
        defaultPanel={defaultPanel}
        disableSubmit
        onCancel={() => {
          ContextMenu.hide();
        }}
      />
    </div>
  );
};

export const showMapStationDetailsPopover = (
  station: any,
  clientX: number,
  clientY: number
): void => {
  ContextMenu.show(
    <MapStationDetails
      stationName={station.name}
      latitude={station.coordinates.latitude}
      longitude={station.coordinates.longitude}
      elevation={station.coordinates.elevation}
      entityType={station.type}
      detailedType={station.statype}
    />,
    {
      left: clientX,
      top: clientY
    },
    undefined,
    true
  );
};
