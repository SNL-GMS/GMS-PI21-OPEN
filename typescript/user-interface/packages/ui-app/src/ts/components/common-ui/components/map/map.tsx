/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import { IS_MODE_IAN } from '@gms/common-util';
import { useUiTheme } from '@gms/ui-state';
import type { Viewer as CesiumViewer } from 'cesium';
import {
  buildModuleUrl,
  Cartesian3,
  Color,
  EllipsoidTerrainProvider,
  KeyboardEventModifier,
  ProviderViewModel,
  ScreenSpaceEventType,
  TileMapServiceImageryProvider
} from 'cesium';
import React from 'react';
import type { CesiumComponentRef } from 'resium';
import {
  Camera,
  CameraFlyTo,
  ImageryLayer,
  Scene,
  ScreenSpaceEvent,
  ScreenSpaceEventHandler,
  Viewer
} from 'resium';

import { MIN_HEIGHT } from './constants';
import { baseViewerSettings } from './map-resources';
import { useOnMultipleSelect, useRenderMapOnSelection, useTileLoadingStatus } from './map-util';
import type { MapProps } from './types';

/* Set to true to see map fps */
const DEBUG_FPS = false;

const imageryProviderViewModels: ProviderViewModel[] = [];

imageryProviderViewModels.push(
  new ProviderViewModel({
    name: 'Natural Earth',
    tooltip: 'Natural Earth',
    iconUrl: buildModuleUrl('Widgets/Images/ImageryProviders/naturalEarthII.png'),
    creationFunction: (): TileMapServiceImageryProvider =>
      new TileMapServiceImageryProvider({
        url: buildModuleUrl('Assets/Textures/NaturalEarthII'),
        maximumLevel: 2
      })
  })
);

const naturalEarthImageryProvider = new TileMapServiceImageryProvider({
  url: buildModuleUrl('Assets/Textures/NaturalEarthII'),
  maximumLevel: 2
});

const terrainProviderViewModels = [];

const wgs84Terrain = new ProviderViewModel({
  name: 'WGS84 Ellipsoid',
  iconUrl: buildModuleUrl('Widgets/Images/TerrainProviders/Ellipsoid.png'),
  tooltip: 'WGS84 standard ellipsoid, also known as EPSG:4326',
  creationFunction: () => new EllipsoidTerrainProvider()
});

terrainProviderViewModels.push(wgs84Terrain);

/**
 * common base map component
 *
 * @param props the props
 */
export function MapComponent(props: MapProps) {
  const ref = React.useRef<CesiumComponentRef<CesiumViewer>>(null);
  useRenderMapOnSelection(ref);

  const { doMultiSelect, selectedStations, selectedEvents, selectedSdIds } = props;

  const minHeight = MIN_HEIGHT;

  // eslint-disable-next-line @typescript-eslint/no-magic-numbers
  const destination: Cartesian3 = Cartesian3.fromDegrees(0, 0, 40000000);

  /**
   * This function passes all the ScreenSpaceEventHandlers
   * viewer context and then maps them into the viewer
   *
   * @returns jsx ScreenSpaceEventHandlers
   */
  const getViewerHandlers = () => {
    return (
      ref?.current?.cesiumElement &&
      props.handlers?.map(handler => handler({ viewer: ref?.current?.cesiumElement }))
    );
  };

  const tileLoadingState = useTileLoadingStatus(ref?.current?.cesiumElement);

  // Guarantee that after the map-panel rerenders, that the map scene also updates with it
  React.useEffect(() => {
    if (IS_MODE_IAN) ref.current?.cesiumElement?.scene?.requestRender();
  });

  // If the user selects a Station using Ctrl-Click, keep the existing Entities and add the new one.
  const onMultipleSelect = useOnMultipleSelect(
    ref,
    selectedStations,
    selectedEvents,
    selectedSdIds
  );

  /*
   * Disable double-click object selection in the UI.
   * NOTE: code added inline here because sonar lint complained about
   * "using a return value from a Void function" when placed in a utility function.
   */
  ref?.current?.cesiumElement?.screenSpaceEventHandler.removeInputAction(
    ScreenSpaceEventType.LEFT_DOUBLE_CLICK
  );

  const [uiTheme] = useUiTheme();

  return (
    <div
      className="map-sub-wrapper"
      style={{ minHeight }}
      data-tile-loading-status={tileLoadingState}
    >
      <Viewer
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...baseViewerSettings}
        ref={ref}
        full
        infoBox={false}
        selectionIndicator={false}
        homeButton
        terrainProviderViewModels={terrainProviderViewModels}
        baseLayerPicker
        imageryProvider={false}
        imageryProviderViewModels={imageryProviderViewModels}
      >
        <Scene
          debugShowFramesPerSecond={DEBUG_FPS}
          backgroundColor={Color.fromCssColorString(uiTheme.colors.gmsBackground)}
        />
        <Camera defaultZoomAmount={0} />
        <CameraFlyTo duration={0} destination={destination} once />
        <ImageryLayer imageryProvider={naturalEarthImageryProvider} />
        {props.entities?.map(entity => {
          return entity;
        })}
        {props.dataSources?.map(dataSource => {
          return dataSource;
        })}
        {getViewerHandlers()}
        {doMultiSelect && (
          <ScreenSpaceEventHandler>
            <ScreenSpaceEvent
              action={onMultipleSelect}
              type={ScreenSpaceEventType.LEFT_DOWN}
              modifier={KeyboardEventModifier.SHIFT}
            />
          </ScreenSpaceEventHandler>
        )}
      </Viewer>
    </div>
  );
}

export const Map = React.memo(MapComponent);
