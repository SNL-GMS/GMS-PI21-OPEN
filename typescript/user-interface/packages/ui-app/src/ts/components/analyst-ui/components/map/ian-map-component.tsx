import { IanDisplays } from '@gms/common-model/lib/displays/types';
import { useInterval } from '@gms/ui-state';
import React from 'react';

import type { IANMapComponentProps } from '~analyst-ui/components/map/types';
import { BaseDisplay } from '~common-ui/components/base-display';

import {
  useIsMapSyncedToWaveformZoom,
  useMapNonPreferredEventData,
  useMapPreferredEventData,
  useSignalDetectionForMap,
  useStationData
} from './ian-map-hooks';
import { IANMapPanel } from './ian-map-panel';

/**
 * IAN Map component. Renders a Cesium map and queries for Station Groups
 */
// eslint-disable-next-line react/function-component-definition
export const IANMapComponent: React.FunctionComponent<IANMapComponentProps> = (
  props: IANMapComponentProps
) => {
  const { glContainer } = props;
  const stationData = useStationData();
  const [interval] = useInterval(useIsMapSyncedToWaveformZoom());
  const signalDetectionData = useSignalDetectionForMap(interval);
  const preferredEventData = useMapPreferredEventData();
  const nonPreferredEventData = useMapNonPreferredEventData();
  const [stationCount, setStationCount] = React.useState<number>(0);
  const [signalDetectionCount, setSignalDetectionCount] = React.useState<number>(0);
  const [preferredEventCount, setPreferredEventCount] = React.useState<number>(0);
  const [nonPreferredEventCount, setNonPreferredEventCount] = React.useState<number>(0);

  const stationMount = React.useRef(() => {
    setStationCount(n => n + 1);
  });
  const signalDetectionMount = React.useRef(() => {
    setSignalDetectionCount(n => n + 1);
  });
  const preferredEventMount = React.useRef(() => {
    setPreferredEventCount(n => n + 1);
  });
  const nonPreferredEventMount = React.useRef(() => {
    setNonPreferredEventCount(n => n + 1);
  });
  return (
    <BaseDisplay
      glContainer={glContainer}
      tabName={IanDisplays.MAP}
      className="ian-map-gl-container"
      data-cy="ian-map-container"
      data-station-count={stationCount}
      data-signal-detection-count={signalDetectionCount}
      data-preferred-event-count={preferredEventCount}
      data-non-preferred-event-count={nonPreferredEventCount}
    >
      <IANMapPanel
        stationsResult={stationData}
        stationMount={stationMount.current}
        signalDetections={signalDetectionData}
        signalDetectionMount={signalDetectionMount.current}
        preferredEventsResult={preferredEventData}
        preferredEventMount={preferredEventMount.current}
        nonPreferredEventsResult={nonPreferredEventData}
        nonPreferredEventMount={nonPreferredEventMount.current}
      />
    </BaseDisplay>
  );
};

export const IANMap = React.memo(IANMapComponent);
