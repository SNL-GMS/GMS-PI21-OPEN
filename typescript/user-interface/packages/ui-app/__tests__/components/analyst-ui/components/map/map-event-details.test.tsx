import React from 'react';
import renderer from 'react-test-renderer';

import {
  MapEventDetails,
  showMapEventDetailsPopover
} from '~analyst-ui/components/map/map-event-details';

import {} from '../../../../../src/ts/components/analyst-ui/components/map/map-station-details';

describe('MapEventDetails', () => {
  test('functions are defined', () => {
    expect(MapEventDetails).toBeDefined();
    expect(showMapEventDetailsPopover).toBeDefined();
  });

  it('matches snapshot', () => {
    const component = renderer
      .create(
        <MapEventDetails
          eventTime={0}
          latitude={45}
          longitude={45}
          depth={10}
          workflowStatus="Not Started"
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
