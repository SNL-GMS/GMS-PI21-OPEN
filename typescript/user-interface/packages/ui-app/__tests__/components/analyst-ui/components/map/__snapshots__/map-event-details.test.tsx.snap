// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapEventDetails matches snapshot 1`] = `
<div
  className="map-station-details__container"
>
  <div
    className="form"
  >
    <h6
      className="bp4-heading form__header"
    >
      <div>
        Event Details
      </div>
    </h6>
    <div
      className="form__panel-selector"
    />
    <div
      className="form-body"
    >
      <div
        className="form-label"
      >
        Event Time:
      </div>
      <div
        className="form-value form-value--uneditable form-value--time"
        title=""
      >
        1970-01-01 00:00:00
      </div>
      <div
        className="form-label"
      >
        Lat (°):
      </div>
      <div
        className="form-value form-value--uneditable form-value--time"
        title=""
      >
        45.000
      </div>
      <div
        className="form-label"
      >
        Lon (°):
      </div>
      <div
        className="form-value form-value--uneditable form-value--time"
        title=""
      >
        45.000
      </div>
      <div
        className="form-label"
      >
        Depth (km):
      </div>
      <div
        className="form-value form-value--uneditable form-value--time"
        title=""
      >
        10.000
      </div>
      <div
        className="form-label"
      >
        Workflow Status:
      </div>
      <div
        className="form-value form-value--uneditable"
        title=""
      >
        Not started
      </div>
    </div>
    <div
      className="form__buttons"
    >
      <div
        className="form__buttons--right"
      >
        <button
          className="bp4-button form__button"
          data-cy="cancel-event-details"
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
          <span
            className="bp4-button-text"
          >
            Close
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
